addons/sale_stock/models/sale_order.py

```diff
@@ -94,6 +100,20 @@ def _compute_delivery_status(self):
     def _compute_expected_date(self):
         super(SaleOrder, self)._compute_expected_date()
 
+    @api.depends('picking_ids.products_availability_state')
+    def _compute_late_availability(self):
+        for order in self:
+            order.late_availability = any(
+                picking.products_availability_state == 'late' for picking in order.picking_ids
+            )
+
+    def _search_late_availability(self, operator, value):
+        if operator not in ('=', '!=') or not isinstance(value, bool):
+            return NotImplemented
+
+        picking_ids = self.env['stock.picking'].search([('products_availability_state', operator, 'late')]).ids
```

Comment:
```suggestion
        picking_ids = self.env['stock.picking'].search([('sale_id', '!=', False), ('products_availability_state', operator, 'late')]).ids
```

Reducing the search space by ignoring irrelevant pickings can make your queries faster, especially on database with many pickings. WDYT?